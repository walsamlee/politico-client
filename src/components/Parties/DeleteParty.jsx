/* eslint-disable no-alert */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';

class DeleteParty extends Component {
    deleteHandler = (event) => {
      event.preventDefault();
      if (window.confirm('Are you sure you want to delete party?')) {
        this.props.deleteParty(this.props.partyId, localStorage.getItem('token'));
      }
    }

    render() {
      return (
        <button type="submit" className="btn btn-cart" onClick={this.deleteHandler}>Delete</button>
      );
    }
}

export default DeleteParty;
