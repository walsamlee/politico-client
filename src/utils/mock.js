const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
