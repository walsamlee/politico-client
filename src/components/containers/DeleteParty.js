import { connect } from 'react-redux';
import DeleteParty from '../Parties/DeleteParty';
import partyAction from '../../actions/partyAction';

const mapStateToProps = state => ({
  data: state,
});

const mapDispatchToProps = dispatch => ({
  deleteParty: (partyId, token) => {
    dispatch(partyAction.deleteParty(partyId, token));
  },
});

const DeleteAParty = connect(mapStateToProps, mapDispatchToProps)(DeleteParty);

export default DeleteAParty;
