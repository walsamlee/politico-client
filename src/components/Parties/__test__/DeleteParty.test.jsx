import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import DeleteParty from '../DeleteParty';

const { token } = mock;

describe('<DeleteParty />', () => {
  it('should render DeleteParty component', () => {
    const wrapper = shallow(
      <DeleteParty
        partyId={1}
        token={token}
        deleteParty={jest.fn()}
      />,
    );
    const event = {
      preventDefault: jest.fn(),
    };
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
    wrapper.instance().deleteHandler(event);
  });
});
