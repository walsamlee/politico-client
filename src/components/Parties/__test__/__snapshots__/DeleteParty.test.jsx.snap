// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteParty /> should render DeleteParty component 1`] = `ShallowWrapper {}`;
